packages/form-core/src/FormApi.ts

```diff
@@ -986,6 +974,19 @@ export class FormApi<
       this.store.setState((prev) => ({ ...prev, isSubmitting: false }))
     }
 
+    // Set all fields to blurred
+    this.store.batch(() => {
+      void (
+        Object.values(this.fieldInfo) as FieldInfo<any, TFormValidator>[]
+      ).forEach((field) => {
+        if (!field.instance) return
+
+        if (!field.instance.state.meta.isBlurred) {
+          field.instance.setMeta((prev) => ({ ...prev, isBlurred: true }))
+        }
+      })
+    })
```

Comment:
I'd like to push the fix in this PR forward but I'm missing a bit, why are we setting all fields to `isBlurred: true` on submit?

I see blur as an explicit action that is: user focused the form, then moved the focus away. This isn't necessarily happening for all fields on submit.

Reply:
My idea was: If the fields are implemented to only show the Error Messages after onBlur and the User submits the Form without blurring all field first the Messages wouldn't show up for those Fields.

**But** that could be handled differently on the field by checking if the form was submitted, that would give us as developers a more fine-grained option when to show the errors.

Reply:
You could do something like showing the error if `field.isBlurred || form.isSubmitted` right?
Then maybe we can remove this part, what do you think?